docs/bpftool-cgroup.rst

```diff
@@ -46,92 +46,86 @@ CGROUP COMMANDS
-		  **recvmsg4** call to recvfrom(2), recvmsg(2), recvmmsg(2) for
-		  an unconnected udp4 socket (since 5.2);
-		  **recvmsg6** call to recvfrom(2), recvmsg(2), recvmmsg(2) for
-		  an unconnected udp6 socket (since 5.2);
-		  **sysctl** sysctl access (since 5.2);
-		  **getsockopt** call to getsockopt (since 5.3);
-		  **setsockopt** call to setsockopt (since 5.3);
-		  **getpeername4** call to getpeername(2) for an inet4 socket (since 5.8);
-		  **getpeername6** call to getpeername(2) for an inet6 socket (since 5.8);
-		  **getsockname4** call to getsockname(2) for an inet4 socket (since 5.8);
-		  **getsockname6** call to getsockname(2) for an inet6 socket (since 5.8).
-		  **sock_release** closing an userspace inet socket (since 5.9).
-
-	**bpftool cgroup detach** *CGROUP* *ATTACH_TYPE* *PROG*
-		  Detach *PROG* from the cgroup *CGROUP* and attach type
-		  *ATTACH_TYPE*.
-
-	**bpftool prog help**
-		  Print short help message.
+    bpftool cgroup { show | list } *CGROUP* [effective]
+        List all programs attached to the cgroup *CGROUP*.
+
+        Output will start with program ID followed by attach type, attach flags 
+        and program name.
+
+        If **effective** is specified retrieve effective programs that will
+        execute for events within a cgroup. This includes inherited along with
+        attached ones.
+
+    bpftool cgroup tree [*CGROUP_ROOT*] [effective]
+        Iterate over all cgroups in *CGROUP_ROOT* and list all attached
+        programs. If *CGROUP_ROOT* is not specified, bpftool uses cgroup v2
+        mountpoint.
+
+        The output is similar to the output of cgroup show/list commands: it
+        starts with absolute cgroup path, followed by program ID, attach type,
+        attach flags and program name.
+
+        If **effective** is specified retrieve effective programs that will
+        execute for events within a cgroup. This includes inherited along with
+        attached ones.
+
+    bpftool cgroup attach *CGROUP* *ATTACH_TYPE* *PROG* [*ATTACH_FLAGS*]
+        Attach program *PROG* to the cgroup *CGROUP* with attach type
+        *ATTACH_TYPE* and optional *ATTACH_FLAGS*.
+
+        *ATTACH_FLAGS* can be one of: **override** if a sub-cgroup installs some
+        bpf program, the program in this cgroup yields to sub-cgroup program;
+        **multi** if a sub-cgroup installs some bpf program, that cgroup program
+        gets run in addition to the program in this cgroup.
+
+        Only one program is allowed to be attached to a cgroup with no attach
+        flags or the **override** flag. Attaching another program will release
+        old program and attach the new one.
+
+        Multiple programs are allowed to be attached to a cgroup with **multi**.
+        They are executed in FIFO order (those that were attached first, run
+        first).
+
+        Non-default *ATTACH_FLAGS* are supported by kernel version 4.14 and
+        later.
+
+        *ATTACH_TYPE* can be one of:
```

Comment:
Sorry I should have mentioned earlier: We do need one blank line here, or the list will not be valid in RST and will not be rendered as such by the parse. If you look at the man page generated from the current version, the types are all inlined.

```suggestion
        *ATTACH_TYPE* can be one of:

```

This is maybe why you got an error when wrapping the lines, too.